src/Swarm/Game/Robot.hs

```diff
@@ -509,6 +513,28 @@ instance FromJSONE EntityMap TRobot where
     mkMachine Nothing = Out VUnit emptyStore []
     mkMachine (Just pt) = initMachine pt mempty emptyStore
 
+type instance Index RobotContext = Ctx.Var
```

Comment:
I like the idea of adding `Ixed` and `At` instances for `RobotContext`, but I would tend to put them right next to the definition of `RobotContext`.  Was there a particular reason for having them down here?

Reply:
I'm just a bit lost in big files. Also I think the best placement for everything will get clearer if we clarify the specific types proposed here... 

I'm also not very sure if I was right not to include the contents of the `Store`...

There's another way to approach this particular change, namely by changing `RobotContext` into a product of `Ctx (Typed Value)` (or whatever we agree on) and the `Store`. I do think the three `Ctx`s should go together somehow. Three different maps potentially mean three different sets of `Var`s: do we want them?

(Such a change will remove the need for those instances)

Reply:
If you have any suggestions on how to split the Robot source file to be easier to orient yourself, feel free to chime in:
- #707 :wink: 

I did not mention this file in the list because it is only over 500 loc, but it is still in our top 10. :grinning:

Reply:
> There's another way to approach this particular change, namely by changing RobotContext into a product of Ctx (Typed Value) (or whatever we agree on) and the Store.

Yes, we contemplated this previously, in #190 .  We abandoned it because it got too complex.  But maybe if we try again we can find a simpler way to do it.